ppr-ui/src/utils/utilities.ts

```diff
@@ -84,32 +85,49 @@ export const filterDuplicates = (list: Array<any>, filterBy: string) => {
   })
 }
 
+/** Helper function to remove properties with undefined/null/empty values **/
+const removeEmptyProperties = (obj: any) => {
+  for (const key in obj) {
+    if (obj[key] === undefined || obj[key] === null || obj[key] === '') {
+      delete obj[key]
+    } else if (typeof obj[key] === 'object') {
+      removeEmptyProperties(obj[key])
+    }
+  }
+}
+
+/** Insensitive string comparison helper. */
+export const insensitiveStrCompare = (v1, v2) => !v1?.localeCompare(v2, undefined, { sensitivity: 'accent' })
+
 /**
  * Deeply compares two values, supporting objects, arrays, and case-insensitive string comparison.
  *
- * @param {*} base - The first value to compare.
- * @param {*} current - The second value to compare.
- * @param {*} isCaseSensitive - Flag for case-sensitive string comparison. Defaults to false.
+ * @param base - The first value to compare.
+ * @param current - The second value to compare.
+ * @param isCaseSensitive - Flag for case-sensitive string comparison. Defaults to false.
+ * @param cleanEmptyProperties - Clean emppty/null/undefined properties from comparisons: true by default.
  * @returns {boolean} - Returns true if the values are different, false if they are equal.
  */
-export const deepChangesComparison = (base: BaseDataUnionIF, current: BaseDataUnionIF): boolean => {
-  // Object safety-check
-  const isObject = value => typeof value === 'object' && value !== null
-  // String normalization safety-check: Case Insensitive
-  const caseInsensitiveStringCompare = (val1, val2) => {
-    if (typeof val1 === 'string' && typeof val2 === 'string') {
-      return val1.toUpperCase() !== val2.toUpperCase()
-    }
-    return val1 !== val2
-  }
+export const deepChangesComparison = (
+  base: BaseDataUnionIF,
+  current: BaseDataUnionIF,
+  isCaseSensitive: boolean = false,
+  cleanEmptyProperties: boolean = true
+): boolean => {
 
-  // Main deep change comparison
-  if (isObject(base) && isObject(current)) {
-    const keys1 = Object.keys(base)
-    const keys2 = Object.keys(current)
+  // Remove undefined properties before comparison
+  if (cleanEmptyProperties) {
```

Comment:
Why remove the empty props? For efficiency?

Reply:
No, lodash comparisons will consider keys with empty values.

Reply:
Ok, so what's the purpose of removal?

Reply:
It's OPTIONAL flexibility. 

If you want a useCase, let's say you have empty properties from an initialized store object that you are not concerned about. Would you rather duplicate countless data models to match every possible scenario or provide some flexibility in a function?

Reply:
Yeah, the use case helped me understand it. That's all I wanted. Thanks